File: examples/snippets/modules/ROOT/examples/pekkostreams-scala/src/main/scala/cloudflow/pekkostreamsdoc/DataHttpIngress.scala
package cloudflow.pekkostreamsdoc

// tag::httpIngress[]
import cloudflow.pekkostream._
import cloudflow.pekkostream.util.scaladsl._

import cloudflow.streamlets.avro._
import cloudflow.streamlets._

import org.apache.pekko.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
import JsonSupport._

class DataHttpIngress extends PekkoServerStreamlet {
  val out         = AvroOutlet[Data]("out").withPartitioner(RoundRobinPartitioner)
  def shape       = StreamletShape.withOutlets(out)
  def createLogic = HttpServerLogic.default(this, out)
}
// end::httpIngress[]

File: examples/snippets/modules/ROOT/examples/pekkostreams-java/src/main/java/cloudflow/pekkostreamsdoc/DataHttpIngress.java
package cloudflow.pekkostreamsdoc;

// tag::httpIngress[]
import cloudflow.pekkostream.PekkoServerStreamlet;

import cloudflow.pekkostream.PekkoStreamletLogic;
import cloudflow.pekkostream.util.javadsl.HttpServerLogic;

import cloudflow.streamlets.RoundRobinPartitioner;
import cloudflow.streamlets.StreamletShape;
import cloudflow.streamlets.avro.AvroOutlet;

import org.apache.pekko.http.javadsl.marshallers.jackson.Jackson;

public class DataHttpIngress extends PekkoServerStreamlet {
  AvroOutlet<Data> out =
      AvroOutlet.<Data>create("out", Data.class)
          .withPartitioner(RoundRobinPartitioner.getInstance());

  public StreamletShape shape() {
    return StreamletShape.createWithOutlets(out);
  }

  public PekkoStreamletLogic createLogic() {
    return HttpServerLogic.createDefault(
        this, out, Jackson.byteStringUnmarshaller(Data.class), getContext());
  }
}
// end::httpIngress[]